### ISSUE TITLE
Feature request for SubStrUsedAsStrPosInspection

### ISSUE THREAD

**Issue by fraserredmond** [kalessil/phpinspectionsea#1007]
| Subject    | Details                                                  |
| :--------- | :------------------------------------------------------- |
| Issue type | Feature request (or False-negative)  |
| IDE        | PhpStorm 2018.2.2                          |
| Plugin     | Php Inspections (EA Extended), 3.0.5.1 |

Description or screenshot:
--

I just noticed today that one of our previous developers always got the the first char of a string using character index notation instead of substr, so I've been trying to track these down :-(

In the following code, both if-statements should get a warning, but only the first does.

```
$foo = 'abcd';
if (substr($foo, 0, 1) == 'f') {
    $hasWarning = true;
}
if ($foo[0] == 'f') {
    $hasWarning = false;
}
```

**Comment by kalessil** [comment]
Implemented!
